js/src/forum/index.js

```diff
@@ -20,31 +20,31 @@ extend(ForumApplication.prototype, 'mount', function() {
   }
 });
 
-m.route.mode = 'pathname';
+m.route.prefix = '';
 
-override(m, 'route', function(original, root, arg1, arg2, vdom) {
-  if (arguments.length === 1) {
+// override(m.route, 'set', function(original, root, arg1, arg2, vdom) {
```

Comment:
The goal here is to override the config="m.route" argument (https://github.com/MithrilJS/mithril.js/blob/e3a5f7486a68c98302aee49da0339985bde7dbac/mithril.js#L1646). With mithril 2, we now use patchMithril to replace vnode tags with m.route.Link. However, that is not currently extensible. We might want to bring that out into it's own method on the modified m object.

Reply:
That code is really horrible.

In the Wordpress extension which uses a variation of this, I've moved `_blank` to a `<base target="_parent">` tag in the `head`.

Aren't we able to override Mithril again here and do a count-action when the `route` attr is present?

Reply:
That's a lot of logic to re-implement: we'd have to override all of global.m again, re-implement the logic, and all that. I think it would be simpler to split up patchMithril into smaller, override-able components, and then override one of them here.

Reply:
Here would actually be a benefit of using `m.route.Link` instead of the `a`/`route` hack. Embed could `extend(m.route.Link, 'view')` as well as `extend(LinkButton, 'view')` to achieve the same results as here, as with the combination of LinkButton+Link this would cover all links the current implementation covers, without any need to hack into Mithril's `m()` helper.

Reply:
Well, since the a/route hack converts <a> and <LinkButton> to <m.route.Link> anyways, we could still probably extend m.route.Link.view

Reply:
@askvortsov1 did you want to attempt extending `m.route.Link` or would you like me to try? I still haven't setup a Mithril v2 environment to work on those v2 PRs and don't know if it's worth doing it or if I should wait before everything gets to master.

Reply:
Up to you!